src/app/psat/system-basics/system-basics.component.ts

```diff
@@ -37,25 +39,51 @@ export class SystemBasicsComponent implements OnInit, OnDestroy {
   oldSettings: Settings;
   showUpdateDataReminder: boolean = false;
   showSuccessMessage: boolean = false;
+  assessmentIntegrationState: IntegrationState;
   @ViewChild('settingsModal', { static: false }) public settingsModal: ModalDirective;
 
-  constructor(private settingsService: SettingsService,    private settingsDbService: SettingsDbService, private psatService: PsatService) { }
+  constructor(private settingsService: SettingsService, private settingsDbService: SettingsDbService, private psatService: PsatService) { }
 
   ngOnInit() {
-    this.settingsForm = this.settingsService.getFormFromSettings(this.settings);
-    this.oldSettings = this.settingsService.getSettingsFromForm(this.settingsForm);
-    if (this.assessment.psat.existingDataUnits && this.assessment.psat.existingDataUnits != this.oldSettings.unitsOfMeasure) {
-      this.oldSettings = this.getExistingDataSettings();
-      this.showUpdateDataReminder = true;
-    }
+    this.initSettings();
   }
 
+  ngOnChanges(changes: SimpleChanges) {
+    if (changes.assessmentPsat) {
+      this.setConnectedInventoryWarning();
+    }
+    if (changes.settings) {
+      this.initSettings();
```

Comment:
initSettings will get called a bunch of times here since it's also in ngOnInit(). could do a check for isFirstChange